mmrotate/core/bbox/coder/angle_coder.py

```diff
@@ -105,7 +106,7 @@ def decode(self, angle_preds):
                 Has shape (num_anchors * H * W, coding_len)
 
         Returns:
-            list[Tensor]: Angle offset for each scale level.
+            Tensor: Angle offset for each scale level.
                 Has shape (num_anchors * H * W, 1)
         """
         angle_cls_inds = torch.argmax(angle_preds, dim=1)
```

Comment:
The shape of returns is wrong, i think the return here is (num_anchors * H * W).
In order to support batch input like (B, N, coding_len), the argmax dim can be -1.


In retinanet, csl directly assign the angle into rbox
```
# `rbbox` has shape (N, 5)
# `ang` has shape (N)
rbbox[..., 4] = ang
```
In some angle_branch head, such as Rotated FCOS and Rotated YOLOX, they use concat to get final bbox.
```
# `hbbox` has shape (N, 4)
# `ang` has shape (N, 1)
rbbox = torch.cat([hbbox, ang], dim=-1)
```

I think we can add a `keepdim` arg in decode method, and use like this:
```
angle_cls_inds = torch.argmax(angle_preds, dim=-1, keepdim=keepdim)
```
When keepdim is true, return (N, 1) or (B, N, 1), otherwise (N,) or (B, N)

Reply:
Done.